src/libs/actions/Report.js

```diff
@@ -1201,6 +1133,108 @@ Onyx.connect({
     callback: handleReportChanged,
 });
 
+/**
+ * Deletes a comment from the report, basically sets it as empty string
+ *
+ * @param {Number} reportID
+ * @param {Object} reportAction
+ */
+function deleteReportComment(reportID, reportAction) {
+    const sequenceNumber = reportAction.sequenceNumber;
+
+    const optimisticReportActions = {
+        [sequenceNumber]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+        },
+    };
+    const optimisticReport = {
+        lastMessageText: ReportActions.getLastVisibleMessageText(reportID, {
+            [sequenceNumber]: {
+                ...reportAction,
+            },
+        }),
+    };
+
+    // If the deleted comment is more recent than our last read comment, update the unread count
+    if (sequenceNumber > getLastReadSequenceNumber(reportID)) {
+        optimisticReport.unreadActionCount = Math.max(getUnreadActionCount(reportID) - 1, 0);
+    }
+
+    // List of optmistic data to change
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: optimisticReportActions,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: optimisticReport,
+        },
+    ];
+
+    // On Success remove the message content and pendingAction state
+    const successData = {
+        [sequenceNumber]: {
+            pendingAction: null,
+            message: [{
+                html: '',
+                text: '',
+            }],
+        },
+    };
+
+    // On Error clear the pendingAction state and revert the message
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: {
+                [sequenceNumber]: {
+                    message: reportAction.message,
+                    pendingAction: null,
+                },
+            },
+        },
+
+        // {
+        //     onyxMethod: CONST.ONYX.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+        //     value: {
+        //         lastMessageText:
+        //         unreadActionCount:
+        //     }
+        // }
+    ];
+
+    const parameters = {
+        reportID,
+        clientID: reportAction.clientID,
```

Comment:
I don't think it is yet possible to delete a comment that doesn't exist yet so this can be removed.